### ISSUE TITLE
Dataset Distillation by Matching Training Trajectories

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#198]
# TL; DR

- Optimize a small synthetic dataset so that it provides similar training trajectory (segment in the parameter space) to that of the model trained on full dataset.

# Motivation

- Dataset distillation has various potential applications (continual learning, NAS, privacy), but has mainly been studied theoretically.
- Scale up to practical level by modeling the mid-range trajectory (neither single update nor long-range optimization).

# Formulation

![image](https://user-images.githubusercontent.com/46308407/205892694-7acabbff-55e6-4002-948d-67f1983e4230.png)

- Precompute the model checkpoints from training on full, real dataset (*expert trajectory*, upper bound).
- Randomly sample a checkpoint snapshopt, and train the model with synthetic (to-be-optimized) dataset.
- After N steps of iterations, compare the distance between this model and the expert model at the next time stamp (normalized L2 distance in the parameter space).
![image](https://user-images.githubusercontent.com/46308407/205893119-fcc2424a-f299-49db-840a-4f036eb389c4.png)
- Backpropagate the distance loss to the dataset (pixels).

![image](https://user-images.githubusercontent.com/46308407/205893084-7904f03b-a198-4ddd-97eb-dcfef118adf1.png)

- Make the learning rate (in the inner loop) learnable to compensate for the hyperparameter choices (N steps in the inner loop, M steps between the expert snapshots).
- Later show that having large M and proportionately large N helps.

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/205893709-6fafc566-e7dd-4b5c-9f17-4ae51b4825d6.png)
![image](https://user-images.githubusercontent.com/46308407/205893767-480c25ff-0115-478f-9d0b-5847183e5425.png)
![image](https://user-images.githubusercontent.com/46308407/205893890-13d6475a-94de-4d05-a61e-ce1ee20cb8df.png)
![image](https://user-images.githubusercontent.com/46308407/205893916-7a2dbf63-63ba-48eb-b686-92913a11e16d.png)
![image](https://user-images.githubusercontent.com/46308407/205893937-46fb0137-15ff-4353-ade6-be0f06802278.png)
![image](https://user-images.githubusercontent.com/46308407/205893963-2916b85a-f48f-417a-a71d-d986e7d0fbee.png)

# Inspiration

- Nice intuition on segmenting the training trajectory of a regular model and enforcing the student updates to imitate it. 
- *Speculate the author might have got the idea from RL (episode simulations).*
- Similar to meta-learning literature, this line of research is very computation-friendly. Small network and small dataset (**128** imagenet **subsets**!!)
- I still believe there are much room to improve upon this work, especially in that the authors directly optimize the pixel values of the synthetic image. 
- We could possibly leverage some sort of prior (whether it be generative or discriminative) to make the procedure more efficient.
